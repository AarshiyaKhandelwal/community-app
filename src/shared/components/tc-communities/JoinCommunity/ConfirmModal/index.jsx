/**
 * The modal asking visitor to confirm that he wants to join community where
 * JoinCommunity component is rendered. If visitor is authenticated it shows
 * Join / Cancel buttons that do what they are supposed to do; otherwise it
 * shows Register & Join / Log In & Join / Cancel buttons, first of which get
 * visitor to registration / login flows, with return url making the join to
 * happen automatically.
 */

/* global window */

import config from 'utils/config';
import Modal from 'components/Modal';
import PT from 'prop-types';
import React from 'react';
import './style.scss';

export default function ConfirmModal({
  communityName,
  groupId,
  join,
  resetJoinButton,
  token,
  userId,
}) {
  return (
    <Modal onCancel={resetJoinButton}>
      <div styleName="confirmMsg">
        { userId ? null : (
          <p>You must be a Topcoder member before you can join the {communityName} community.</p>
        )}
        To join, login if you are already a member. If not, register first.
      </div>
      <div styleName="buttons">
        { userId ? (
          <span>
            <button
              onClick={() => join(token, groupId, userId)}
              styleName="btnConfirm"
            >Join</button>
            <button
              onClick={resetJoinButton}
              styleName="btnCancel"
            >Cancel</button>
          </span>
        ) : (
          <span>
            <button
              onClick={() => {
                const url = `${
                  encodeURIComponent(window.location.href)
                }?join=${groupId}`;
                window.location = `${config.URL.AUTH}/member?retUrl=${url}`;
              }}
              styleName="btnConfirmLong"
            >Login</button>
            <button
              onClick={() => {
                const url = `${
                  encodeURIComponent(window.location.href)
                }?join=${groupId}`;
                window.location = `${config.URL.AUTH}/member/registration?retUrl=${url}`;
              }}
              styleName="btnConfirmLong"
            >Register</button>
            <button
              onClick={resetJoinButton}
              styleName="btnCancelLong"
            >Cancel</button>
          </span>
        )}
      </div>
    </Modal>
  );
}

ConfirmModal.defaultProps = {
  token: null,
  userId: null,
};

ConfirmModal.propTypes = {
  communityName: PT.string.isRequired,
  groupId: PT.string.isRequired,
  join: PT.func.isRequired,
  resetJoinButton: PT.func.isRequired,
  token: PT.string,
  userId: PT.string,
};
