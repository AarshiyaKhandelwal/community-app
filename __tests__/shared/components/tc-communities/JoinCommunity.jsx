import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import JoinCommunity from 'components/tc-communities/JoinCommunity';

const hideJoinButton = jest.fn();
const join = jest.fn();
const resetJoinButton = jest.fn();
const showJoinConfirmModal = jest.fn();

const mockDatas = [
  {
    communityName: 'ios',
    groupId: '1',
    hideJoinButton,
    join,
    resetJoinButton,
    showJoinConfirmModal,
    state: 'confirm-join',
    token: 'token',
    userId: 'useId',
  }, {
    communityName: 'ios',
    groupId: '1',
    hideJoinButton,
    join,
    resetJoinButton,
    showJoinConfirmModal,
    state: 'joining',
    token: 'token',
    userId: 'useId',
  }, {
    communityName: 'ios',
    groupId: '1',
    hideJoinButton,
    join,
    resetJoinButton,
    showJoinConfirmModal,
    state: 'hidden',
    token: 'token',
    userId: 'useId',
  }, {
    communityName: 'ios',
    groupId: '1',
    hideJoinButton,
    join,
    resetJoinButton,
    showJoinConfirmModal,
    state: 'joined',
    token: 'token',
    userId: 'useId',
  }, {
    communityName: 'ios',
    groupId: '1',
    hideJoinButton,
    join,
    resetJoinButton,
    showJoinConfirmModal,
    state: 'confirm-join',
    userId: 'useId',
  },
];

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();
  _.forEach(mockDatas, (data) => {
    renderer.render((
      <JoinCommunity {...data} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

class Wrapper extends React.Component {
  componentDidMount() {}
  render() {
    return <JoinCommunity {...this.props} />;
  }
}

describe('click confirm-join', () => {
  const instance = TU.renderIntoDocument((<Wrapper {...mockDatas[0]} />));
  const matches = TU.scryRenderedDOMComponentsWithTag(instance, 'button');
  _.forEach(matches, m => TU.Simulate.click(m));
  expect(join).toHaveBeenCalled();
});


describe('click joined', () => {
  const instance = TU.renderIntoDocument((<Wrapper {...mockDatas[3]} />));
  const matches = TU.scryRenderedDOMComponentsWithTag(instance, 'button');
  TU.Simulate.click(matches[0]);
});

describe('click without token', () => {
  const instance = TU.renderIntoDocument((<Wrapper {...mockDatas[4]} />));
  const matches = TU.scryRenderedDOMComponentsWithTag(instance, 'button');
  TU.Simulate.click(matches[0]);
});
