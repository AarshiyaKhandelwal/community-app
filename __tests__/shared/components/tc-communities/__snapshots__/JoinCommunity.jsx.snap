// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div>
  <button
    className="src-shared-components-tc-communities-JoinCommunity-___style__link___mMDH0"
    onClick={[Function]}
  >
    Join Community
  </button>
  <Modal
    onCancel={[Function]}
  >
    <p
      className="src-shared-components-tc-communities-JoinCommunity-___style__confirmMsg___3Urx0"
    >
      Are you sure you want to join 
      ios
      ?
    </p>
    <div
      className="src-shared-components-tc-communities-JoinCommunity-___style__buttons___24jiR"
    >
      <button
        className="src-shared-components-tc-communities-JoinCommunity-___style__btnConfirm___21w2M"
        onClick={[Function]}
      >
        Join
      </button>
      <button
        className="src-shared-components-tc-communities-JoinCommunity-___style__btnCancel___21ib7"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </Modal>
</div>
`;

exports[`Matches shallow shapshot 2`] = `
<div>
  <button
    className="src-shared-components-tc-communities-JoinCommunity-___style__link___mMDH0 src-shared-components-tc-communities-JoinCommunity-___style__joining___rq-m2"
    onClick={[Function]}
  >
    <div>
      <p>
        Joining...
      </p>
      <ThemedLoadingIndicator
        composeTheme="deeply"
        mapThemrProps={[Function]}
        theme={
          Object {
            "style": "src-shared-components-tc-communities-JoinCommunity-___style__loadingIndicator___2RP1u",
          }
        }
      />
    </div>
  </button>
</div>
`;

exports[`Matches shallow shapshot 3`] = `
<div
  className="src-shared-components-tc-communities-JoinCommunity-___style__placeholder___1XfW2"
/>
`;

exports[`Matches shallow shapshot 4`] = `
<div>
  <button
    className="src-shared-components-tc-communities-JoinCommunity-___style__link___mMDH0"
    onClick={[Function]}
  >
    Join Community
  </button>
  <Modal
    onCancel={[Function]}
  >
    <h1
      className="src-shared-components-tc-communities-JoinCommunity-___style__modalTitle___1U_iw"
    >
      Congratulations!
    </h1>
    <p
      className="src-shared-components-tc-communities-JoinCommunity-___style__modalMsg___25f_a"
    >
      You have joined 
      ios
      !
    </p>
    <button
      className="src-shared-components-tc-communities-JoinCommunity-___style__done___2m1SD"
      onClick={[Function]}
    >
      Return to the Community
    </button>
  </Modal>
</div>
`;

exports[`Matches shallow shapshot 5`] = `
<div>
  <button
    className="src-shared-components-tc-communities-JoinCommunity-___style__link___mMDH0"
    onClick={[Function]}
  >
    Join Community
  </button>
  <Modal
    onCancel={[Function]}
  >
    <p
      className="src-shared-components-tc-communities-JoinCommunity-___style__confirmMsg___3Urx0"
    >
      Are you sure you want to join 
      ios
      ?
    </p>
    <div
      className="src-shared-components-tc-communities-JoinCommunity-___style__buttons___24jiR"
    >
      <button
        className="src-shared-components-tc-communities-JoinCommunity-___style__btnConfirm___21w2M"
        onClick={[Function]}
      >
        Join
      </button>
      <button
        className="src-shared-components-tc-communities-JoinCommunity-___style__btnCancel___21ib7"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </Modal>
</div>
`;
